Update configure-otel-collector.md: update recommendation for OTEL (#9769)
grafana/mimir: docs/sources/mimir/configure/configure-otel-collector.md

# Configure the OpenTelemetry Collector to write metrics into Mimir

{{% admonition type="note" %}}
To send OTel data to Grafana Cloud, refer to [Send data using OpenTelemetry Protocol (OTLP)](https://grafana.com/docs/grafana-cloud/send-data/otlp/send-data-otlp/).
{{% /admonition %}}

When using the [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/), you can write metrics into Mimir via two options: `prometheusremotewrite` and `otlphttp`.

> We recommend using each protocol's respective exporter and native Mimir endpoint:

Reviewer: ```suggestion
As a best practice, use each protocol's respective exporter, as well as the following Mimir endpoints:
```

Reviewer: I tried to clear this up a bit. LMK is the meaning is preserved, or if we should try something else.

Author: I would leave this as is. "We recommend" is the language we use elsewhere in the docs, and I think "Best Practice" implies that a lot more research and consensus than we want to imply here.

Author: also the endpoints aren't explicitly listed here, those names are for the exporters.
The endpoints are expanded on in each corresponding section below.
We could drop the bit about the endpoints completely, but saying "the following Mimir endpoints" would not make sense.

Reviewer: Thanks for clarifying. Let's leave as it is 😄

> - Prometheus data through `prometheusremotewrite`

Reviewer: ```suggestion
- For Prometheus data, `prometheusremotewrite`
```

Author: I don't think bullet points need to be worded so close to a full proper sentence, starting it with a prepositional phrase. I do not see other examples of us taking this approach in the docs. 

The word "through" could just be a colon if we want to shorten up

Reviewer: Yea, let's make it a colon, it cleans it up a bit.

> - OTel data through `otlphttp`

Reviewer: ```suggestion
- For OTel data, `otlphttp`
```

When using the [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/), you can write metrics into Mimir via two options: `prometheusremotewrite` and `otlphttp`.

> We recommend using each protocol's respective exporter and native Mimir endpoint.

Reviewer 2: Not sure exactly how to interpret this. Why not just recommend to use the `otlphttp` exporter, as it's now our recommendation?

```suggestion
We recommend using the `otlphttp` exporter when possible.
```

Author: The OTEL Collector and distributions like Alloy can also receive Prometheus metrics, and export Prometheus metrics. The article is titled "Configure the OTEL Collector", it does not appear to be exclusive to sending OTEL data. So Prometheus should go through Prometheus, and OTEL through OTEL.

We can get more specific, I left it off at first because I thought it sounded repetitive:

```markdown
We recommend using each protocol's respective exporter and native Mimir endpoint:
* Promtheus data through `prometheusremotewrite`
* `OTLP` data through `otlphttp`

Reviewer 2: Thanks for the explanation, I don't think I was aware of OTel Collector being able to receive Prometheus metrics. I think I prefer your more explicit version.

> `OTLP` data through `otlphttp`

I think "OTel data through `otlphttp`" makes more sense, since the data itself is OTel format and OTLP is just the wire protocol.

Author: updated
